### ISSUE TITLE
Validation is adding new object in form-reducer

### ISSUE THREAD

**Issue by obykoo** [davidkpiano/react-redux-form#688]
I do not know if I am doing something wrong ,but I spent whole day debugging only to come to conlusion that validation is adding new object to form reducer data.

Lets say I have this model:

```
const initData = {
 someArray=[]
}

const store = applyMiddleware(thunk)(createStore)(combineReducers({
    data: modelReducer('data', initData ),
    dataForm: formReducer('data', initData ),
}));
```

Model is used in Form:

```
<Form model="data"
   onSubmit={(data) => this.handleSubmit(data)}
   onSubmitFailed={ (dataForm) => this.handleSubmitFailed(dataForm) }
 >
{antoher components..}
</Form>
```

Then I have mapped someArray to Control components:

```
<Control
   model={"data.someArray["+parseInt(i)+"]"}
   component={CustomControlComponent}
   mapProps={{
       passedProps: (props) => props,
        value: (props) => props.viewValue
     }}
    key={data.someArray[parseInt(i)].ID}   //ID is generated unique for each object
   validators={{
      something: (value) => this.checkSomeArray(value),
    }}
  validateOn="change"
/>

  checkSomeArray(object){
        if (object.dataA==null) {
            return false;
        }
        if (object.dataB==null) {
            return false;
        }
        return true;
    }
```

Here is scernario I follow:

**1. Add empty object to array** 
```
let obj={
 dataA:null,
 dataB:null
}
dispatch(actions.push("data.someArray",obj));
```
Result:
```
model:
 [0]: { dataA:null , dataB:null }

form (invalid):
[0]: { dataA:null , dataB:null } , invalid
```

**2. Change some data in object**

`dispatch(actions.change("data.someArray[0].dataA","somethingA"));`
`dispatch(actions.change("data.someArray[0].dataB","somethingB"));`

Result:
```
model:
 [0]: { dataA:"somethingA" , dataB:"somethingB"l }

form (valid) :
 [0]: { dataA:"somethingA" , dataB:"somethingB"l } , valid
```

**3. Add another empty object to array**
```
let obj={
 dataA:null,
 dataB:null
}
dispatch(actions.push("data.someArray",obj));
```

Result:
```
model:
 [0]: { dataA:"somethingA" , dataB:"somethingB"l }
 [1]: { dataA:null , dataB:null }

form (invalid) :
 [0]: { dataA:"somethingA" , dataB:"somethingB"l } , valid
 [1]: { dataA:null , dataB:null } invalid
```

4. Delete object with null data

`dispatch(actions.remove("data.someArray",1));`

Result:
```
model:
 [0]: { dataA:"somethingA" , dataB:"somethingB" }

form (invalid) :
 [0]: { dataA:"somethingA" , dataB:"somethingB" } , valid
 [1]: { null } valid
```
Basicly after deleting object , new object with value null appear in form data , but not in model data.
This object is also not validated, but is valid. However whole array is stated as invalid.

When I replace validator in form with:

`  'someArray[]': (value) => true;`

this does not happen.

**Comment by obykoo** [comment]
It happening only probably somewhere in my code , its more complicated then this example. But how do I debug this change?

**Comment by obykoo** [comment]
[https://github.com/davidkpiano/react-redux-form/issues/619](619) .. this seems like exactly same issue I am having.

I tried to add "scenario"  from second post to codepen , but scenario is working correctly.
However I have little bit more complicated Form then this simple scenario ,but workflow is almost same.

**Comment by davidkpiano** [comment]
Closing as this seems to be fixed in #619. Please let me know if you still have the same problem with the latest version.
